cmd/podman/images/scp.go

```diff
@@ -0,0 +1,324 @@
+	sessions[1].Stdout = &buffer
+	sessions[1].Stderr = &bufferErr
+	if err := sessions[1].Run(run); err != nil { // run the command declared above to create new temp file
+		return "", errors.Wrapf(err, bufferErr.String())
+	}
+	n, err := scpD.CopyFrom(dial, remoteFile, localFile)
+	if err != nil {
+		errOut := (strconv.Itoa(int(n)) + " Bytes copied before error")
+		return " ", errors.Wrapf(err, errOut)
+	}
+	return "", nil
+}
+
+// parseLocation returns the valid connection data based off of the information provided by the user
+// iden is a string array of ssh identity keys, url is a string array of paths to ssh clients. The first entry
+// in each array is the save location and the second entry is the load location if applicable. An empty string will be
+// inserted if identity is blank (not required in some cases).
+// serv is a map of service destinations with the connection name as the index intended to be used as EngineConfig.ServiceDestinations
+// this function modifies the global scpOpt entities: FromRemote, ToRemote, Connections, and DestImageName
+func parseArgs(args []string, cfg *config.Config) ([]string, []string, map[string]config.Destination, error) {
+	iden := []string{}
+	url := []string{}
+	serv := map[string]config.Destination{}
+	cliConnections := []string{}
+	switch len(args) {
+	case 1:
+		if strings.Contains(args[0], "::") {
+			scpOpts.FromRemote = true
+			cliConnections = append(cliConnections, args[0])
+		} else {
+			scpOpts.DestImageName = args[0]
+		}
+	case 2:
+		if strings.Contains(args[0], "::") {
+			if !(strings.Contains(args[1], "::")) { // if an image is specified, this mean we are loading from our client
+				cliConnections = append(cliConnections, args[0])
+				scpOpts.ToRemote = true
+			} else if strings.Contains(args[1], "::") { // both remote clients
+				scpOpts.FromRemote = true
+				scpOpts.ToRemote = true
+				if len(strings.Split(args[0], "::")) == 1 || (len(strings.Split(args[0], "::")) == 2 && len(strings.Split(args[1], "::")) == 2) { // is save->load w/ one image name or save-load w/ two image names
+					cliConnections = append(cliConnections, args[0])
+					cliConnections = append(cliConnections, args[1])
+				} else { // else its a load save (order of args)
+					cliConnections = append(cliConnections, args[1])
+					cliConnections = append(cliConnections, args[0])
+				}
+			}
+		} else if strings.Contains(args[1], "::") && len(cliConnections) < 2 { // if we are given image host::
+			scpOpts.DestImageName = args[0]
+			scpOpts.ToRemote = true
+			cliConnections = append(cliConnections, args[1])
+		}
+	default:
+		return nil, nil, nil, errors.Wrapf(define.ErrInvalidArg, "expected at least 1 argument")
+	}
+	for i, val := range cliConnections {
+		splitEnv := strings.SplitN(val, "::", 2)
+		switch len(splitEnv[1]) == 0 {
+		case true:
+			scpOpts.Connections = append(scpOpts.Connections, splitEnv[0])
+		case false:
+			scpOpts.Connections = append(scpOpts.Connections, splitEnv[0])
```

Comment:
I'm dizzy from following the back-and-forth here, but I caught this one just by accident: line 309 is identical to 307, which means it should be refactored outside the conditional.

And, like @mtrmac said, please do not use `switch` for a true/false.

Reply:
thanks @edsantiago will change now